microsoft/testsuites/mdatp/mdatp.py

```diff
@@ -22,6 +24,70 @@
 EXIT_ONBOARD_INFO_FOUND = 255
 
 
+def apply_working_path_noexec_workaround(node: Node) -> None:
+    # EXAMPLE WORKAROUND:
+    # - Get the mount point for the path.
+    # - Check if the mount point is mounted noexec.
+    # - if yes, remount with exec flag
+    working_path = node.get_working_path()
+    mount_info = node.tools[Df].get_mount_info_for_dir(
+        directory=working_path.as_posix()
+    )
+    if mount_info and check_noexec_partition(node, mount_info):
+        node.log.info(f"Working path in {mount_info.mountpoint} is mounted noexec!!")
+        remount_without_noexec(node, mount_info)
+    elif not mount_info:
+        # warn only, since nearly all (99.99%) of images do not
+        # require this workaround. An image with weird DF or Mount
+        # output should still be able to run the script, so we
+        # will warn and allow the test to continue.
+        node.log.warn(
+            f"Could not locate mount info for directory "
+            f"{working_path.as_posix()}. "
+            "Test may fail due to noexec permissions error. "
+            "Or due to the working path not existing."
+        )
+
+    # no issues with partition, nothing to fix.
+    return
+
+
+def check_noexec_partition(node: Node, partition: PartitionInfo) -> bool:
+    # an example. I don't have time to implement this in mount,
+    # we currently discard the mount options so they're not exposed
+    # in tools.Mount
+    for part in node.tools[Mount].get_partition_info():
+        node.log.debug(f"checking partition {str(part)}...")
+        if part.mount_point == partition.mountpoint:
+            return "noexec" in part.options
+    # This path would mean that df and mount are returning different
+    # info about the mount points on the VM.
+    # There should be no way for this to happen, assert if it does
+    node.log.warn(
+        f"Could not find partition info for {partition.mountpoint}! "
+        "This indicates that LISA could not properly parse the output "
+        "of 'mount' and 'df'. It's possible this image does not have "
+        "these tools available. "
+        "Note: This is a non-fatal error for the mdatp test. "
+        "Resuming text execution..."
+    )
+    # appease pylint by returning after assert
+    return False
+
+
+def remount_without_noexec(node: Node, partition: PartitionInfo) -> None:
+    # Fix issue where working path is created in a partition marked
+    # 'noexec' meaning file permissions for x are ignored.
+    # Attempt to remount the parition to allow executables.
+    # The other fix would just be adding a disk, creating a parition
+    # marking it executable, and using that as the working path.
+    node.tools[Mount].mount(
+        options="remount,exec",
```

Comment:
How about existing other options, like "local" in the example? It may need to get all options, and remove "noexec", and then add "remount" and "exec".

Reply:
refactored to handle bsd/linux remount arg differences and preserve options on both